cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationRegistrationStage.java

```diff
@@ -118,7 +132,7 @@ public boolean apply(Map.Entry<String, ScheduleSpecification> input) {
                                                newSpecification.getFlows(), newSpecification.getMapReduce(),
                                                newSpecification.getSpark(), newSpecification.getWorkflows(),
                                                newSpecification.getServices(), filteredExistingSchedules,
-                                               newSpecification.getProgramSchedules(),
+                                               filteredExistingProgramSchedules,
```

Comment:
this was not the point of my comment. The new app spec must be consistent in that it does not define schedules for programs that don't exist. I assume that is checked somewhere else. Which means that this filetring won't filter anything out. 

However, somewhere you need to delete the schedules from the schedule store. In the ScheduleDeletionStage you will not do that if canUpdateSchedules is false. That was original comment: even if it is false, it needs to delete these program schedules. Not from the app spec, but from the scheduler.